### ISSUE TITLE
TravisCI reconfiguration

### ISSUE THREAD

**Issue by bastimeyer** [streamlink/streamlink#2128]
We need to reconfigure the TravisCI hooks, because GH services have been deprecated since the beginning of this month and TravisCI is still registered as one of these here in the Streamlink repo. Old integrations will be removed at the end of January 2019 and need to be replaced with webhooks.
https://developer.github.com/changes/2018-04-25-github-services-deprecation/

What needs to be done now?
Will simply removing and adding TravisCI again work? What will happen to the build history? I couldn't find any answers to these questions in the TravisCI docs. Maybe I'm just blind...

Since the TravisCI GH service was registered by @gravyboat when Streamlink was forked, I decided to not mess with it on my own and open this thread instead.

----

Also regarding CI tools, I've seen that Windows is now supported by TravisCI.
https://blog.travis-ci.com/2018-10-11-windows-early-release

Python environments don't seem to be available in the Windows VMs/containers yet, but once they become available, I'd say that we should switch and remove AppVeyor, as it is redundant. AppVeyor is so utterly slow and bad, replacing it could speed up the overall CI process. It's ridiculous that several builds are still queued since almost an entire day after the database issues of GH yesterday, even though the issue has been resolved and the webhooks have been re-triggered. TravisCI by comparison has caught up quickly in just a few hours.

Also, I've noticed that the Streamlink builds on AppVeyor are registered on @beardypig's account. It's not really a problem, but other people don't have access to the account, so this might cause problems in the future if we don't intend to remove the AppVeyor integration.

**Comment by beardypig** [comment]
👍 TravisCI for Windows. I cannot see how to make the project as part of an organisation on AppVeyor...

**Comment by bastimeyer** [comment]
https://docs.travis-ci.com/user/open-source-on-travis-ci-com/#existing-open-source-repositories-on-travis-ciorg

> Open source projects and their build history will continue to run on travis-ci.org for now.
> 
> However, open source repositories will be migrated to travis-ci.com gradually. The migration was planned to start at the end of Q2 2018, but has been pushed back. We will announce a new date as soon as we are able. You will receive an email when the migration for a repository is complete. This is an opt-in process: to have a repository migrated over, it must first be activated on travis-ci.com.
> 
> Currently, open source repositories that were private in any time in their history may be added to travis-ci.com. The option to migrate to travis-ci.com for active open source repositories that were always open source will come soon.
> 
> Repositories may also be migrated without their build history or build settings (including environment variables) immediately.
> 
> Please contact support support@travis-ci.com. In your email, please include:
> 
> - "Repository Migration" somewhere in the subject line, and
> - the repository’s name in your email

**Comment by bastimeyer** [comment]
@beardypig 
I've contacted AppVeyor earlier today because of the long queue times. The builds should be working again now. They also suggested turning on rolling builds in the config menu on their site (can't be added in the YAML file) which can now be enabled for PRs specifically.
https://www.appveyor.com/docs/build-configuration/#rolling-builds
This will make https://github.com/streamlink/streamlink/blob/master/appveyor.yml#L16-L24 redundant...

**Comment by bastimeyer** [comment]
So as it seems, the TravisCI webhook was added automatically a couple of days ago:
https://blog.travis-ci.com/2019-01-31-github-services-integration-deprecation
The TravisCI "Github service" can be removed from the repo settings now (or not, since it's deactivated anyway):
https://github.com/streamlink/streamlink/settings/installations
TL;DR: no user interaction required :grinning: 

----

Regarding Windows builds on TravisCI, I think we should slowly transition to it, because Appveyor is failing randomly again. I know it's not officially supported yet, but python choco packages are available in various versions.

As an example, this config file was linked in one of the threads on the Travis support forum:
https://github.com/cclauss/itinerant-tester-windows/blob/d4edf42ef52b270fc77e1a0e7d989b9336ad5868/.travis.yml#L6-L22

And the remaining dependencies like nsis, imagemagick and inkscape should also be available.

**Comment by back-to** [comment]
> And the remaining dependencies like nsis, imagemagick and inkscape should also be available.

as this is only needed for `./script/makeinstaller.sh` ,
should this script also work on windows?

---

this diff will run travis with windows,
but **script**, **after_success**, **before_deploy** and **deploy** needs an update depends what should also run on windows or/and linux

```diff
diff --git a/.travis.yml b/.travis.yml
index a664f15f..e19d819b 100644
--- a/.travis.yml
+++ b/.travis.yml
@@ -10,31 +10,67 @@ git:
 
 matrix:
   include:
-  - python: '2.7'
-  - python: '3.4'
-  - python: '3.5'
+  - os: linux
+    python: '2.7'
+  - os: linux
+    python: '3.4'
+  - os: linux
+    python: '3.5'
     env: BUILD_DOCS=yes BUILD_INSTALLER=yes BUILD_SDIST=yes DEPLOY_PYPI=yes
-  - python: '3.6'
-  - python: '3.7'
+  - os: linux
+    python: '3.6'
+  - os: linux
+    python: '3.7'
     dist: xenial
     sudo: true
-  - python: '3.8-dev'
+  - os: linux
+    python: '3.8-dev'
     dist: xenial
     sudo: true
+  - os: windows
+    python: '2.7'
+    before_install:
+      - choco install python2
+      - export PATH="/c/Python27:/c/Python27/Scripts:$PATH"
+    language: shell
+  - os: windows
+    python: '3.4'
+    language: shell
+    before_install:
+      - choco install python --version 3.4.4.20180111
+      - export PATH="/c/Python34:/c/Python34/Scripts:$PATH"
+  - os: windows
+    python: '3.5'
+    # env: BUILD_DOCS=yes BUILD_INSTALLER=yes
+    language: shell
+    before_install:
+      - choco install python --version 3.5.4
+      - export PATH="/c/Python35:/c/Python35/Scripts:$PATH"
+  - os: windows
+    python: '3.6'
+    language: shell
+    before_install:
+      - choco install python --version 3.6.8
+      - export PATH="/c/Python36:/c/Python36/Scripts:$PATH"
+  - os: windows
+    python: '3.7'
+    language : shell
+    before_install:
+      - choco install python
+      - export PATH="/c/Python37:/c/Python37/Scripts:$PATH"
   allow_failures:
   - python: '3.8-dev'
 
-before_install:
-  - pip install --disable-pip-version-check --upgrade pip setuptools
-  - pip install --upgrade -r dev-requirements.txt
-  - pip install pycountry
+install:
+  - python --version
+  - python -m pip install --disable-pip-version-check --upgrade pip setuptools
+  - python -m pip install --upgrade -r dev-requirements.txt
+  - python -m pip install pycountry
   - if [[ $BUILD_DOCS == 'yes' ]]; then
-      pip install -r docs-requirements.txt;
-      pip install doctr;
+      python -m pip install -r docs-requirements.txt;
+      python -m pip install doctr;
     fi
-
-install:
-  - pip install -e .
+  - python -m pip install -e .
 
 script:
   - pytest --cov
```

**Comment by bastimeyer** [comment]
> should this script also work on windows?

No... Just forget what I said there, I wasn't thinking.

> **script**, **after_success**, **before_deploy** and **deploy** needs an update

**Comment by bastimeyer** [comment]
misclicked... will edit my comment above

**Comment by back-to** [comment]
here with `TRAVIS_OS_NAME`

https://github.com/streamlink/streamlink/compare/master...back-to:windows
https://travis-ci.org/back-to/sl/builds/491253953

**Comment by back-to** [comment]
https://github.com/streamlink/streamlink/pull/2656
https://github.com/streamlink/streamlink/pull/2860
https://github.com/streamlink/streamlink/pull/2862
